proto/examples/sasltest/main.rs

```diff
@@ -0,0 +1,167 @@
+use core::panic;
+use futures_util::future::err;
+use sspi::builders::EmptyInitializeSecurityContext;
+use sspi::AuthIdentity;
+use sspi::ClientRequestFlags;
+use sspi::CredentialUse;
+use sspi::DataRepresentation;
+use sspi::Ntlm;
+use sspi::SecurityBuffer;
+use sspi::SecurityBufferType;
+use sspi::SecurityStatus;
+use sspi::Sspi;
+use sspi::SspiImpl;
+use sspi::Username;
+use std::env;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use tokio::net::TcpStream;
+use tokio_util::codec::Framed;
+use tracing::Level;
+
+use futures_util::sink::SinkExt;
+use futures_util::stream::StreamExt;
+
+use ldap3_proto::proto::*;
+use ldap3_proto::LdapCodec;
+
+#[tokio::main]
+async fn main() -> Result<(), ()> {
+    let subs = tracing_subscriber::FmtSubscriber::builder()
+        .with_max_level(Level::DEBUG)
+        .finish();
+    tracing::subscriber::set_global_default(subs).expect("setting default subscriber failed");
+
+    let ldap_password = env::var("LDAP_PASSWORD").unwrap(); // password
+    let ldap_server_addr = env::var("LDAP_SERVER_ADDR").unwrap(); // domain.com:port
+    let ldap_username = env::var("LDAP_USERNAME").unwrap(); // username@domain
+    let addr = SocketAddr::from_str(&ldap_server_addr).expect(&format!(
+        "Unable to parse address, addr is {:?}",
+        &ldap_server_addr
+    ));
+
+    let tcpstream = TcpStream::connect(addr)
+        .await
+        .map_err(|e| eprintln!("Failed to connect to -> {:?}", e))?;
+
+    let mut framed = Framed::new(tcpstream, LdapCodec::default());
+
+    let mut ntlm = AuthProvier::new(&ldap_username, &ldap_password);
+    let ntlm_token = ntlm.step(&[]).unwrap();
+
+    let msg = LdapMsg {
+        msgid: 1,
+        op: LdapOp::BindRequest(LdapBindRequest {
+            dn: "".to_string(),
+            cred: LdapBindCred::SASL(SaslCredentials {
+                mechanism: "GSS-SPNEGO".to_string(),
+                credentials: ntlm_token,
+            }),
+        }),
+        ctrl: vec![],
+    };
+
+    let _ = framed.send(msg).await.map_err(|e| {
+        eprintln!("Unable to send bind -> {:?}", e);
+    })?;
+    loop {
+        if let Some(Ok(msg)) = dbg!(framed.next().await) {
+            if let LdapOp::BindResponse(res) = dbg!(msg.op) {

```

Comment:
Question: is leaving the `dbg!` intentional? Typically, logging using tracing would be preferred (especially given that tracing is already used otherwise).